xbmc/video/ContextMenus.cpp

```diff
@@ -168,4 +171,26 @@ bool CPlay::Execute(const CFileItemPtr& itemIn) const
   return true;
 };
 
+
+std::string CFavorite::GetLabel(const CFileItem& item) const
+{
+  //The 0 we're passing for contextWindow should only matter to folders
+  return g_localizeStrings.Get(XFILE::CFavouritesDirectory::IsFavourite(std::make_shared<CFileItem>(item).get(), 0) ? 14077 : 14076); // Add/Remove Favourite
```

Comment:
XFILE::CFavouritesDirectory::IsFavourite also returns true for a favourite iself. After #11929 got merged, this will lead now to an item "Remove from favourites" and an additional item "Remove" for every favourite. 

This can be corrected by changing the condition here to 

<code>!URIUtils.IsProtocol(item.GetPath(), "favourites") && XFILE::CFavouritesDirectory::IsFavourite(std::make_shared<CFileItem>(item).get()</code>